tests/integration/tools/atlas/atlasHelpers.ts

```diff
@@ -143,3 +144,61 @@ async function createProject(apiClient: ApiClient): Promise<Group & Required<Pic
 
     return group as Group & Required<Pick<Group, "id">>;
 }
+
+export function sleep(ms: number): Promise<void> {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+export async function deleteAndWaitCluster(
+    session: Session,
+    projectId: string,
+    clusterName: string,
+    pollingInterval: number = 1000
+): Promise<void> {
+    await session.apiClient.deleteCluster({
+        params: {
+            path: {
+                groupId: projectId,
+                clusterName,
+            },
+        },
+    });
+    while (true) {
```

Comment:
We shouldn't do this because it can block the entire MCP Server if there is an issue in Atlas or it doesn't work as we expected.

We should:

1. Always provide an upper found. It can be relatively high, like 100 iterations, but must be unbound.
2. Provide an abort signal, so the caller can stop waiting for the deleteCluster (for example, requests timeouts can cancel this).


```suggestion
    for (let try = 0; try < MAX_ITERATIONS && !abortSignal.aborted; try++)
```

AbortSignal would be a parameter provided by the caller in deleteAndWaitCluster.

Reply:
Yes, we can definitely add an upper bound for tries here. Do we need the abortSignal for now? We don't have any external triggers where we need to pass in an AbortController.